src/frontend/qt_sdl/RAMInfoDialog.cpp

```diff
@@ -0,0 +1,302 @@
+/*
+    Copyright 2016-2021 Arisotura
+
+    This file is part of melonDS.
+
+    melonDS is free software: you can redistribute it and/or modify it under
+    the terms of the GNU General Public License as published by the Free
+    Software Foundation, either version 3 of the License, or (at your option)
+    any later version.
+
+    melonDS is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+    FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with melonDS. If not, see http://www.gnu.org/licenses/.
+*/
+
+#include "RAMInfoDialog.h"
+#include "ui_RAMInfoDialog.h"
+
+#include "main.h"
+
+extern EmuThread* emuThread;
+
+s32 GetMainRAMVAlue(const u32& addr, const ramInfo_ByteType& byteType)
+{
+    switch (byteType)
+    {
+    case ramInfo_OneByte:
+        return ((s8*)NDS::MainRAM)[addr&NDS::MainRAMMask];
+    case ramInfo_TwoBytes:
+        return ((s16*)NDS::MainRAM)[addr&NDS::MainRAMMask];
```

Comment:
isn't this the wrong way around? Since addr is in bytes, it should be `*(s16*)(&NDS::MainRAM[addr&NDS::MainRAMMask])`?

Reply:
Oh, I was like a stupid. 
You're right. I'll fix it right away